Fix non-root containers with home directory in non-existent subdirectories of

meyerj opened the pull request:
`useradd -m` cannot handle that case (https://unix.stackexchange.com/a/365361), and the previous merge conflict resolution with osrf/rocker#71 (run `mkhomedir_helper`) was invalid (duplicate home directory creation).

meyerj committed:
Merge branch 'master' into fix-non-root-containers

meyerj committed:
Merge remote-tracking branch 'origin/master' into fix-non-root-containers

meyerj committed:
Merge remote-tracking branch 'Intermodalics/master' into fix-non-root-containers

fevb commented:
It doesn't yet work for me on the beast:
```bash
> /opt/benchmarking/imvps-master/docker/run.sh
...
building > Step 5/6 : RUN mkhomedir_helper francisco.vina
building >  ---> Running in c944bec093ab
no more output and success not detected
Build failed exiting
```
With the --home option it does seem to work though

meyerj committed:
Create the parent directory of the home directory before calling mkhomedir_helper

The tool fails with exit code 6 if the top-level directory does not exist yet.

    (a39bc32231c8)root@c71fe951b480:/# getent passwd johannes
    johannes:x:1000006:1000:Johannes Meyer:/share/homes/johannes:/bin/bash
    (a39bc32231c8)root@a39bc32231c8:/# rm share -R
    (a39bc32231c8)root@a39bc32231c8:/# mkhomedir_helper johannes; echo $?
    6
    (a39bc32231c8)root@a39bc32231c8:/# mkdir share
    (a39bc32231c8)root@a39bc32231c8:/# mkhomedir_helper johannes; echo $?
    0

meyerj committed:
Merge remote-tracking branch 'meyerj/fix-non-root-containers' into HEAD

laurent-engels reviewed (approved):
LGTM
